trade-finance-manager-ui/server/helpers/deal-cancellation-enabled.test.ts

```diff
@@ -0,0 +1,55 @@
+import * as ukefCommon from '@ukef/dtfs2-common';
+import { DEAL_SUBMISSION_TYPE } from '@ukef/dtfs2-common';
+import { dealCancellationEnabled } from './deal-cancellation-enabled';
+
+describe('dealCancellationEnabled', () => {
+  const isTfmDealCancellationFeatureFlagEnabledSpy = jest.spyOn(ukefCommon, 'isTfmDealCancellationFeatureFlagEnabled');
+
+  describe('when TFM Deal Cancellation feature flag is enabled', () => {
+    beforeEach(() => {
+      isTfmDealCancellationFeatureFlagEnabledSpy.mockReturnValue(true);
+    });
+
+    it('should return false if the deal type is MIA', () => {
+      const result = dealCancellationEnabled(DEAL_SUBMISSION_TYPE.MIA);
+
+      expect(result).toEqual(false);
+    });
+
+    it('should return true if the deal type is MIN', () => {
+      const result = dealCancellationEnabled(DEAL_SUBMISSION_TYPE.MIN);
+
+      expect(result).toEqual(true);
+    });
+
+    it('should return true if the deal type is AIN', () => {
+      const result = dealCancellationEnabled(DEAL_SUBMISSION_TYPE.AIN);
+
+      expect(result).toEqual(true);
+    });
+  });
+
+  describe('when TFM Deal Cancellation feature flag is disabled', () => {
+    beforeEach(() => {
+      isTfmDealCancellationFeatureFlagEnabledSpy.mockReturnValue(false);
+    });
+
+    it('should return false if the deal type is MIA', () => {
+      const result = dealCancellationEnabled(DEAL_SUBMISSION_TYPE.MIA);
+
+      expect(result).toEqual(false);
+    });
+
+    it('should return false if the deal type is MIN', () => {
+      const result = dealCancellationEnabled(DEAL_SUBMISSION_TYPE.MIN);
+
+      expect(result).toEqual(false);
+    });
+
+    it('should return false if the deal type is AIN', () => {
+      const result = dealCancellationEnabled(DEAL_SUBMISSION_TYPE.AIN);
+
+      expect(result).toEqual(false);
+    });
```

Comment:
We could combine these all in an `it.each([MIA,MIN,AIN])` :)

Reply:
We absolutely could, have now done that thanks! :)